File: Classes/EditablePageCellTextField.m
// FuelCalculatorTextField.m
//
// Kraftstoff


#import "EditablePageCellTextField.h"


@implementation EditablePageCellTextField

- (instancetype)initWithFrame:(CGRect)frame
{
    if ((self = [super initWithFrame:frame])) {

        _allowCut   = NO;
        _allowPaste = NO;
    }

    return self;
}


// Disable Cut&Paste functionality to properly handle special text inputs methods for our textfields
- (BOOL)canPerformAction:(SEL)action withSender:(id)sender
{
    if (action == @selector(cut:) && !self.allowCut)
        return NO;

    if (action == @selector(paste:) && !self.allowPaste)
        return NO;

    return [super canPerformAction:action withSender:sender];
}

@end

File: Classes/EditablePageCellTextField.swift
//
//  EditablePageCellTextField.swift
//  kraftstoff
//
//  Created by Ingmar Stein on 03.05.15.
//
//

import UIKit

class EditablePageCellTextField: UITextField {
  var allowCut = false
  var allowPaste = false

  // Disable cut & paste functionality to properly handle special text inputs methods for our textfields
  override func canPerformAction(_ action: Selector, withSender sender: Any?) -> Bool {
    if action == #selector(UIResponderStandardEditActions.cut(_:)), !allowCut {
      return false
    }

    if action == #selector(UIResponderStandardEditActions.paste(_:)), !allowPaste {
      return false
    }

    return super.canPerformAction(action, withSender: sender)
  }
}
